messaging/artemis-jta/src/main/java/io/quarkus/ts/openshift/messaging/artemisjta/PriceResource.java

```diff
@@ -0,0 +1,92 @@
+package io.quarkus.ts.openshift.messaging.artemisjta;
+
+import javax.inject.Inject;
+import javax.jms.JMSException;
+import javax.transaction.Transactional;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import static io.quarkus.ts.openshift.messaging.artemisjta.ConsumerService.exToS;
+
+@Path("/")
+public class PriceResource {
+
+    @Inject
+    ConsumerService c;
+
+    @Inject
+    ProducerService p;
+
+    @POST
+    @Path("/price")
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Transactional
+    public Response postCustom(@QueryParam("fail") boolean fail, @NotNull String price) {
+        try {
+            p.produceCustomPrice(price, fail);
```

Comment:
JAX-RS resource methods can return normal values and throw exceptions. If there's no `ExceptionMapper` for the exception, it will end up as server error (500), which is exactly what you're doing here manually.